cli/cmd/compliance.go

```diff
@@ -263,3 +283,67 @@ func buildComplianceReportTable(detailsTable, summaryTable, recommendationsTable
 	}
 	return mainReport.String()
 }
+
+func filterRecommendations(recommendations []api.ComplianceRecommendation) ([]api.ComplianceRecommendation, string) {
+	var filtered []api.ComplianceRecommendation
+	for _, r := range recommendations {
+		if matchRecommendationsFilters(r) {
+			filtered = append(filtered, r)
+		}
+	}
+	if len(filtered) == 0 {
+		return filtered, "There are no recommendations with the specified filter(s).\n"
+	}
+
+	cli.Log.Debugw("filtered recommendations", "recommendations", filtered)
+	return filtered, fmt.Sprintf("%v of %v recommendations showing \n", len(filtered), len(recommendations))
+}
+
+func matchRecommendationsFilters(r api.ComplianceRecommendation) bool {
+	var results []bool
+
+	// severity returns specified threshold and above
+	if compCmdState.Severity != "" {
+		sevThreshold, _ := eventSeverityToProperTypes(compCmdState.Severity)
```

Comment:
nit: rename `eventSeverityToProperTypes()` to `severityToProperTypes()` since it is now used in multiple commands